util/chunk/alloc.go

```diff
@@ -123,32 +123,33 @@ func (alloc *poolColumnAllocator) put(col *Column) {
 
 	l := alloc.pool[typeSize]
 	if l == nil {
-		l = &freeList{
-			data: make([]*Column, 0, 8),
-		}
+		l = make(map[*Column]struct{}, 8)
 		alloc.pool[typeSize] = l
 	}
 	l.push(col)
 }
 
-type freeList struct {
-	data []*Column
-}
+// freeList is defined as a map, rather than a list, because when recycling chunk
+// columns, there could be duplicated one: some of the chunk columns are just the
+// reference to the others.
+type freeList map[*Column]struct{}
```

Comment:
Does this map affects the performance improvements?

Reply:
It's allocated once and reused...